### PR TITLE
Add exponential backoff for deletion failures.

### PR DESCRIPTION
**Pull request by tseaver**
Fixes: #1657.

Note for potential broader use toward #1619.

### CODE REVIEW

File: system_tests/logging_.py [hunk 1]
```diff
@@ -50,8 +50,18 @@ def setUp(self):
...
     def tearDown(self):
+        from gcloud.exceptions import NotFound
         for doomed in self.to_delete:
-            doomed.delete()
+            backoff_intervals = [1, 2, 4, 8]
+            while True:
+                try:
+                    doomed.delete()
+                    break
+                except NotFound:
```
Comment (dhermes) [inline system_tests/logging_.py]: Have you seen this occur in the wild? Why would a `NotFound` occur on something that needs to be deleted? Also, wouldn't it occur once the delete succeeds?

Comment (tseaver) [inline system_tests/logging_.py]: > Have you seen this occur in the wild? Why would a NotFound occur on something that needs to be deleted?

#1657 is the `NotFound`-for-just-created-logs issue.

> Also, wouldn't it occur once the delete succeeds?

The `break` exits the `while` loop if the `delete()` call doesn't error out.

Comment (dhermes) [inline system_tests/logging_.py]: But if `doomed` is already deleted before the first `doomed.delete()` then it'll just error out 4 times.

As for #1657, can we hold off on a more conclusive answer before implementing?

Comment (tseaver) [inline system_tests/logging_.py]: > But if doomed is already deleted before the first doomed.delete() then it'll just error out 4 times.

Things in `doomed` were created by the test being torn down, which won't have deleted them already.

> As for #1657, can we hold off on a more conclusive answer before implementing?

This change addresses @munangst's theory that eventual consistency is what has been causing the 404's.  Do you have any other theory?  With this fix, I have all tests passing (again, after they began failing again after the brief happy interval earlier).

**Comment by dhermes** [comment]
LGTM

**Pull request merged**